pkg/kubelet/pod_workers_test.go

```diff
@@ -414,13 +414,15 @@ func TestUpdatePodWithTerminatedPod(t *testing.T) {
 		RunningPod: runningPod,
 	})
 
+	drainAllWorkers(podWorkers)
+
 	if podWorkers.IsPodKnownTerminated(pod.UID) == true {
 		t.Errorf("podWorker state should not be terminated")
 	}
 	if podWorkers.IsPodKnownTerminated(terminatedPod.UID) == false {
 		t.Errorf("podWorker state should be terminated")
 	}
-	if podWorkers.IsPodKnownTerminated(runningPod.ID) == true {
+	if podWorkers.IsPodKnownTerminated(runningPod.ID) == false {
```

Comment:
This change doesn't make sense to me. I expect the runningPod to not be known terminated

Reply:
"to not be known terminated"... is the logic change here. We expect the running pod to be terminated, if it is not terminated then fail the test.